src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,1543 @@
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>React
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Modernizr, jQuery, jQuery UI
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Modernizr, jQuery, jQuery Migrate, jQuery UI
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Modernizr, jQuery, jQuery Migrate
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Lodash, jQuery
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Lightbox, jQuery, jQuery Migrate
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Lightbox, jQuery
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>Isotope, jQuery, jQuery Migrate
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>GSAP, Lodash, Polyfill, React, RequireJS, jQuery
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>GSAP, Lodash, Polyfill, React, jQuery
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>FancyBox, jQuery, jQuery UI
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td>FancyBox, jQuery
+   </td>
+   <td><p style="text-align: right">
+1%</p>
+
+   </td>
+  </tr>
+</table>
+
+<figcaption>{{ figure_link(caption="Javascript frameworks usage with JavaScript libraries.", sheets_gid="1934429143", sql_file="frameworks_libraries_combos.sql") }}</figcaption>
+</figure>
+
+As can be observed, jQuery tops in usage with libraries and frameworks, which could also be a result of using a third-party library that includes it. This, as a result, leads to more processing time for the code that the framework or library uses with the other included libraries.
+
+22% of known versions of jQuery are found to be version 3.5.1. This is a big jump compared to last year’s (1.12.4), which could be attributed to Wordpress which constitutes most of the participation of jQuery. 
+
+**Web components and shadow DOM**
+
+With the web becoming componentized, a developer building a single page application may think about a user view as a set of components. This is not only for the sake of developers building dedicated components for each feature, but also to maximize component reusability. It could be in the same app on a different view or in a completely different application. Such use cases lead to the usage of custom elements and web components in web applications. 
+
+It would be justified to say that with many JavaScript frameworks gaining popularity, the idea of reusability and building dedicated feature-based components has been adopted more widely.  This feeds our curiosity to look into the adoption of custom elements, shadow DOM, template elements.
```

Comment:
```suggestion
It would be justified to say that with many JavaScript frameworks gaining popularity, the idea of reusability and building dedicated feature-based components has been adopted more widely. This feeds our curiosity to look into the adoption of custom elements, shadow DOM, template elements.
```